file-formats/cfdf/type/zif_aff_cfdf_v1.intf.abap

```diff
@@ -0,0 +1,589 @@
+INTERFACE zif_aff_cfdf_v1
+  PUBLIC.
+
+  TYPES: ty_field_name              TYPE c LENGTH 30.
+  TYPES: ty_component_identifier    TYPE c LENGTH 5.
+  TYPES: ty_business_context        TYPE c LENGTH 30.
+  TYPES: ty_code_value              TYPE c LENGTH 30.
+  TYPES: ty_data_subject_id_type    TYPE c LENGTH 10.
+  TYPES: ty_type                    TYPE c LENGTH 10.
+  TYPES: ty_message_direction       TYPE c LENGTH 3.
+  TYPES: ty_field_control_property  TYPE c LENGTH 1.
+  TYPES: ty_aggregation             TYPE c LENGTH 5.
+
+  TYPES:
+    "! <p class="shorttext">Label and Tooltip</p>
+    "! Label and tooltip
+    BEGIN OF ty_gs_ui_text,
+      "! <p class="shorttext">Translation Language</p>
+      "! Translation language
+      "! $required
+      language      TYPE sy-langu,
+
+      "! <p class="shorttext">Field Label</p>
+      "! Field label
+      "! $required
+      "! $maxLength 40
+      field_label   TYPE c LENGTH 40,
+
+      "! <p class="shorttext">Field Tooltip</p>
+      "! Field tooltip
+      "! $required
+      "! $maxLength 60
+      field_tooltip TYPE c LENGTH 60,
+    END OF ty_gs_ui_text.
+  TYPES: ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text WITH KEY language.
+
+  TYPES:
+    "! <p class="shorttext">Extended Business Context Details</p>
+    "! Extended business context details
+    BEGIN OF ty_gs_extended_bus_context,
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
+      business_context     TYPE ty_business_context,
+
+      "! <p class="shorttext">Deviating Field Name</p>
+      "! Deviating field name
+      "! $maxLength 30
+      deviating_field_name TYPE ty_field_name,
+    END OF ty_gs_extended_bus_context.
+  TYPES: ty_gt_extended_bus_context TYPE STANDARD TABLE OF ty_gs_extended_bus_context WITH KEY business_context.
+
+  TYPES:
+    "! <p class="shorttext">Extended CDS View Details</p>
+    "! Extended CDS view details
+    BEGIN OF ty_gs_extended_cds_view,
+      "! <p class="shorttext">CDS View Name</p>
+      "! CDS view name
+      "! $required
+      "! $maxLength 30
+      cds_view_name                 TYPE zif_aff_types_v1=>ty_object_name_30,
+
+      "! <p class="shorttext">Business Context</p>
+      "! Business context
+      "! $required
+      "! $maxLength 30
+      business_context              TYPE ty_business_context,
+
+      "! <p class="shorttext">Basic Search Relevance</p>
+      "! Basic search relevance
+      is_search_relevant            TYPE abap_bool,
+
+      "! <p class="shorttext">BAdI Implementation for Visibility Control</p>
+      "! BAdI implementation for visibility control
+      "! $maxLength 30
+      badi_impl_for_visibility_ctrl TYPE zif_aff_types_v1=>ty_object_name_30,
+
+      "! <p class="shorttext">Field Control Property</p>
+      "! Field control property
+      "! $maxLength 1
```

Comment:
In my opinion, maxLength 1 is wrong here. It is not necessary, since it is an enum valued field. Furthermore, the values of the enum have length longer than 1 ("optional",...). 
Maybe for your understanding: maxLength is the maximum length for the JSON value (e.g. "optional") and not the ABAP value (e.g. " ")

Reply:
Thanks for pointing this out. In fact I was not aware of this. All occurrences of $maxLength were removed...